### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Tejinder15**
# Added Authentication

## Things to check :
1. Whether you are able to access cart Page, wishlist Page without login
2. whether you are able to successfully login or not
3. whether you are able to logout after logging in

### CODE REVIEW

File: src/Components/Header/Header.jsx [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+    const logoutHandler = () => {
+        redirect("/");
+        localStorage.removeItem("token");
+        localStorage.removeItem("user");
+        authDispatch({type:"LOGOUT"});
+    }
+
+    const userHandler = async (type) => {
+        type === "Login" ? redirect("/login"): logoutHandler();
+    }
```
Comment (Abhijeet43) [inline src/Components/Header/Header.jsx]: Hey @Tejinder15 can you just tell is there any specific reason of using async here?

**Review by Abhijeet43 — CHANGES_REQUESTED** [review]
Change the pointer on you login button to cursor

**Review by Abhijeet43 — APPROVED** [review]
The Code is working well.. Code structure is also good

**Review by mohammed-sas — APPROVED** [review]
1.auth context has been written well
2.functionalities are working as desired
3.overall code is fine

**Pull request merged**